README.md
code-array
----------
[![Build Status](https://travis-ci.org/jarst/code-array.svg?branch=master)](https://travis-ci.org/jarst/code-array)
[![Coverage Status](https://coveralls.io/repos/github/jarst/code-array/badge.svg)](https://coveralls.io/github/jarst/code-array)
[![npm:](https://img.shields.io/npm/v/code-array.svg)](https://www.npmjs.com/packages/code-array)
[![license:mit](https://img.shields.io/badge/license-mit-green.svg)](#license)


Simple utility module that will help you with converting:
 * strings to character code arrays
 * character code arrays to strings

 ## Installation
 `npm install code-array --save`

 ## Example usage

```
 var codeArray = require('code-array');

 var arr = codeArray.fromString('ABC'); // arr is  [65, 66, 67]

 var str = codeArray.toString([65, 66, 67]); // str is 'ABC'
 ```


CHANGES.md
# Change Log

## [0.0.2] - 2016-08-02
### Changed
Tests and build configs added to .npmignore

## [0.0.1] - 2016-08-31
### Initial release

test/test-code-array.js
/**
 * Created by jarst on 2016-08-31.
 */
var chai = require('chai');
var codeArray = require('./../index');

var should = chai.should();

describe('codeArray', function() {
    describe('fromString', function () {
        it('should convert string to code array', function () {
            var str = 'BAR';
            var result = codeArray.fromString(str);

            result.should.be.eql([66, 65, 82]);
        });
        it('should convert string object to code array', function () {
            var str = new String('a');
            var result = codeArray.fromString(str);

            result.should.be.eql([97]);
        });
        it('should return empty array for empty string', function() {
            var result = codeArray.fromString('');

            result.should.be.eql([]);
        });
        it('should return empty array for no argument', function() {
            var result = codeArray.fromString();

            result.should.be.eql([]);
        });
        it('should return empty array for non strings objects', function() {
            var notAString = {'0': 'a', length: 1};
            var result = codeArray.fromString(notAString);

            result.should.be.eql([]);
        });
    });

    describe('toString', function () {
        it('should convert codes array to string', function () {
            var codes = [66, 65, 82];
            var result = codeArray.toString(codes);

            result.should.be.equal('BAR');
        });
        it('should convert codes array object to string', function () {
            var codes = new Array([97]);
            var result = codeArray.toString(codes);

            result.should.be.equal('a');
        });
        it('should return empty string for empty array', function() {
            var result = codeArray.toString([]);

            result.should.be.equal('');
        });
        it('should return empty array for no argument', function() {
            var result = codeArray.toString();

            result.should.be.equal('');
        });
        it('should return empty string for non arrays', function() {
            var notAnArray = {'0': '65', length: 1};
            var result = codeArray.toString(notAnArray);

            result.should.be.equal('');
        });
        it('should ignore non-number values in codes array', function() {
            var codes = [97, 'b', '99', '100foo'];
            var result = codeArray.toString(codes);

            result.should.be.equal('ac');
        });

    });

});


lib/code-array.js
/**
 * Created by jarst on 2016-08-31.
 */

function isString(o) {
    return typeof o === 'string' || o instanceof String;
}

/**
 * Converts String to Array of UTF-16 character codes.
 * @param {string} str
 * @returns {Number[]}
 */
function stringToCodeArray(str) {
    var codes = [];
    if (isString(str)) {
        for (var i = 0; i < str.length; ++i) {
            codes.push(str.charCodeAt(i));
        }
    }
    return codes;
}

/**
 * Converts Array of UTF-16 character codes to String.
 * @param {Number[]} codes
 * @returns {string}
 */
function codeArrayToString(codes) {
    var str = '';
    if (Array.isArray(codes)) {
        for (var i = 0; i < codes.length; ++i) {
            var code = +codes[i];
            if (code) {
                str += String.fromCharCode(code);
            }
        }
    }
    return str;
}

module.exports = {
    fromString: stringToCodeArray,
    toString: codeArrayToString
};

test-code-array.js
/**
 * Created by jarst on 2016-08-31.
 */
var chai = require('chai');
var codeArray = require('./code-array');

var should = chai.should();

describe('codeArray', function() {
    describe('fromString', function () {
        it('should convert string to code array', function () {
            var str = 'BAR';
            var result = codeArray.fromString(str);

            result.should.be.eql([66, 65, 82]);
        });
        it('should convert string object to code array', function () {
            var str = new String('a');
            var result = codeArray.fromString(str);

            result.should.be.eql([97]);
        });
        it('should return empty array for empty string', function() {
            var result = codeArray.fromString('');

            result.should.be.eql([]);
        });
        it('should return empty array for no argument', function() {
            var result = codeArray.fromString();

            result.should.be.eql([]);
        });
        it('should return empty array for non strings objects', function() {
            var notAString = {'0': 'a', length: 1};
            var result = codeArray.fromString(notAString);

            result.should.be.eql([]);
        });
    });

    describe('toString', function () {
        it('should convert codes array to string', function () {
            var codes = [66, 65, 82];
            var result = codeArray.toString(codes);

            result.should.be.equal('BAR');
        });
        it('should convert codes array object to string', function () {
            var codes = new Array([97]);
            var result = codeArray.toString(codes);

            result.should.be.equal('a');
        });
        it('should return empty string for empty array', function() {
            var result = codeArray.toString([]);

            result.should.be.equal('');
        });
        it('should return empty array for no argument', function() {
            var result = codeArray.toString();

            result.should.be.equal('');
        });
        it('should return empty string for non arrays', function() {
            var notAnArray = {'0': '65', length: 1};
            var result = codeArray.toString(notAnArray);

            result.should.be.equal('');
        });
        it('should ignore non-number values in codes array', function() {
            var codes = [97, 'b', '99', '100foo'];
            var result = codeArray.toString(codes);

            result.should.be.equal('ac');
        });

    });

});
